### PR TITLE
feat: private route

### PR DESCRIPTION
**Pull request by fedasevich**
## Describe your changes

- Added private routing that includes
1. Logged in users
2. Verified users
3. Two user roles
- If any of the conditions are not met, the user will be redirected to the correct page

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/CC-183?atlOrigin=eyJpIjoiYTY4ZWFhNWZmODk1NGM1OTk0MGVmMTEzMjRjMGQxYmUiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/private-route/constants.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { USER_ROLE } from 'src/constants';
+import { ROUTES } from 'src/routes';
+
+export const DEFAULT_REDIRECT_PATH = ROUTES.login;
+
+export const REDIRECT_PATHS = {
+  [USER_ROLE.Caregiver]: ROUTES.schedule,
+  [USER_ROLE.Seeker]: ROUTES.create_appointment,
```
Comment (nadiia-tsytsylina) [inline src/components/private-route/constants.ts:8]: here we redirect user only to main page. we don't have schedule and appointments pages

File: src/components/auth-provider/AuthProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { useAppDispatch, useTypedSelector } from 'src/redux/store';
+
+interface AuthProviderProps {
+  children: React.ReactNode;
+}
+
+export function AuthProvider({ children }: AuthProviderProps): JSX.Element | null {
+  const dispatch = useAppDispatch();
+
+  const [loading, setLoading] = useState(true);
```
Comment (Lilly-nn) [inline src/components/auth-provider/AuthProvider.tsx:13]: types

File: src/components/private-route/PrivateRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export function PrivateRoute({
+  children,
+  allowedRole,
+  isIndex,
+}: PrivateRouteProps): JSX.Element | null {
+  const router = useRouter();
+  const pathname = usePathname();
+
+  const [isAllowedToEnter, setIsAllowedToEnter] = useState(false);
```
Comment (Lilly-nn) [inline src/components/private-route/PrivateRoute.tsx:23]: types

File: src/components/private-route/constants.ts (hunk 1 above)
Comment (fedasevich) [reply src/components/private-route/constants.ts:6]: fixed

File: src/components/auth-provider/AuthProvider.tsx (hunk 2 above)
Comment (fedasevich) [reply src/components/auth-provider/AuthProvider.tsx]: fixed

File: src/components/private-route/PrivateRoute.tsx (hunk 3 above)
Comment (fedasevich) [reply src/components/private-route/PrivateRoute.tsx]: fixed

**Review by ProggerOleg — APPROVED** [review]

**Review by Simpsonkk — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**